crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -318,12 +318,18 @@ impl Settlement {
         self.boundary.solver
     }
 
-    /// The settled user orders.
-    pub fn orders(&self) -> HashSet<order::Uid> {
+    /// The settled user orders with their in/out amounts.
+    pub fn orders(&self) -> HashMap<order::Uid, competition::OrderAmounts> {
         self.solutions
             .values()
-            .flat_map(|solution| solution.user_trades().map(|trade| trade.order().uid))
-            .collect()
+            .fold(Default::default(), |mut acc, solution| {
+                for trade in solution.user_trades() {
+                    let order = acc.entry(trade.order().uid).or_default();
+                    order.in_amount = trade.order().target().into();
```

Comment:
To me, in amount is always the sell amount, whereas `target()` seems to differentiate between buy and sell orders.

Reply:
Also, should this account for fee (since we only have a single clearing price, it would be nice to see for each order their net (after fee) in/out amounts?

Reply:
At this point I'm unsure what `in_amount` and `out_amount` are exactly.
Should they refer to `sell_amount`/`buy_amount` or `executed_amount`/`received_amount`? In any case it looks to me like aggregating `.executed()` in the `out_amount` is a bug as this is the `buy_amount` for buy orders or `sell_amount` for sell orders.

Reply:
I understood it is a `executed_amount/received_amount` but now I am not sure if that makes sense, or what we actually want to have.

Reply:
> In any case it looks to me like aggregating .executed() in the out_amount is a bug as this is the buy_amount for buy orders or sell_amount for sell orders.

Based on my understanding ☝️ , it's not bug since both `in_amount` and `out_amount` are buy values(what we sent vs what solver executed) for `buy` orders and sell values (what we sent vs what solver executed) for `sell` orders.

Question remains if this is what we want, but with this understanding, solver competition is populated as it was before.

Reply:
I think we should store "how much did the user sell", "how much did the user buy" since from a user perspective this is what they care about (bang for buck). The fact that we have executed amount right now is confusing in my opinion (if 10% was executed in two solutions I still don't know which solution would have been better for me as a user), so while we may have to support it for now, I think we should prepare for more intuitive amount reporting in the solver competition endpoint.